extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/identity/Oauth2ServiceImpl.java

```diff
@@ -74,25 +75,17 @@
      * @param client                    Http client
      * @param jwtDecoratorRegistry      Registry containing the decorator for build the JWT
      * @param typeManager               Type manager
-     * @param additionalValidationRules An optional list of {@link ValidationRule} that are evaluated <em>after</em> the
-     *                                  standard OAuth2 validation
      */
-    public Oauth2ServiceImpl(Oauth2Configuration configuration,
-                             JWSSigner tokenSigner,
-                             OkHttpClient client,
-                             JwtDecoratorRegistry jwtDecoratorRegistry,
-                             TypeManager typeManager,
-                             JwtClaimValidationRule... additionalValidationRules) {
+    public Oauth2ServiceImpl(Oauth2Configuration configuration, JWSSigner tokenSigner, OkHttpClient client, JwtDecoratorRegistry jwtDecoratorRegistry, TypeManager typeManager) {
         this.configuration = configuration;
         this.typeManager = typeManager;
         httpClient = client;
         this.jwtDecoratorRegistry = jwtDecoratorRegistry;
         this.tokenSigner = tokenSigner;
 
-        var rules = new ArrayList<JwtClaimValidationRule>();
-        rules.add(new Oauth2ValidationRule(configuration.getProviderAudience(), this.configuration)); //OAuth2 validation must ALWAYS be done
-        rules.addAll(List.of(additionalValidationRules));
-        this.tokenValidationService = new TokenValidationServiceImpl(configuration.getIdentityProviderKeyResolver(), Collections.unmodifiableList(rules));
+        //OAuth2 validation must ALWAYS be done
+        validationRules = List.of(new Oauth2ValidationRule(this.configuration));
```

Comment:
`List.of()` creates an immutable list, which will cause Line 98 to raise an `UnsupportedOperationException`. Clearly this has not been tested.

Reply:
Done (please check again if ok)